### DISCUSSION TITLE
Use codedeploy for deploy new version of service

### DISCUSSION THREAD

**Discussion by flemzord** [aws/copilot-cli#2872] [Q&A]
Hello, 

Do you plan to support deployment via CodeDeploy to take advantage of BlueGreen etc? 
(I didn't find any link to the RoadMap?)

Regards, 
Maxence

**Comment by efekarakus** [comment]
Hi @flemzord !

We'd love to bring blue/green support into Copilot it's one of our top most voted issues: https://github.com/aws/copilot-cli/issues/1758 but unfortunately the CloudFormation support today is very limited: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/blue-green.html#blue-green-considerations (which is needed for Copilot).

> Declaring output values or importing values from other stacks isn't currently supported for templates defining blue/green ECS deployments. 

> You can't use the AWS::CodeDeploy::BlueGreen hook in a nested stack. 

These make the feature incompatible today with Copilot 😭. I know that the ECS team is well aware of these limitations and whenever these restrictions are lifted, we'll support it in Copilot!
